enterprise/server/remote_execution/containers/podman/podman.go

```diff
@@ -42,9 +52,17 @@ var (
 
 	memUsagePathTemplate = flag.String("executor.podman.memory_usage_path_template", "/sys/fs/cgroup/memory/libpod_parent/libpod-{{.ContainerID}}/memory.usage_in_bytes", "Go template specifying a path pointing to a container's current memory usage, in bytes. Templated with `ContainerID`.")
 	cpuUsagePathTemplate = flag.String("executor.podman.cpu_usage_path_template", "/sys/fs/cgroup/cpuacct/libpod_parent/libpod-{{.ContainerID}}/cpuacct.usage", "Go template specifying a path pointing to a container's total CPU usage, in CPU nanoseconds. Templated with `ContainerID`.")
-	// TODO(iain): switch to boolean flag once image streaming supports all container images.
-	streamableImages = flagutil.New("executor.podman.streamable_images", []string{}, "List of images that can be streamed by podman.")
-	pullTimeout      = flag.Duration("executor.podman.pull_timeout", 10*time.Minute, "Timeout for image pulls.")
+
+	// TODO(iain): delete the streamableImages flag
+	imageStreamingEnabled = flag.Bool("executor.podman.enable_image_streaming", false, "If set, all podman images are streamed using soci artifacts generated and stored in the apps.")
+	streamableImages      = flagutil.New("executor.podman.streamable_images", []string{}, "List of podman images that can be streamed using registry-stored soci artifacts. Note that if executor.podman.enable_image_streaming is set then all images are streamed using app-stored soci artifacts and the value of this flag is ignored.")
+
+	pullTimeout = flag.Duration("executor.podman.pull_timeout", 10*time.Minute, "Timeout for image pulls.")
+
+	// Currently, the app_internal target is only used for the
+	// SociArtifactStore, so its definition and use is restricted to this file.
```

Comment:
You can make this a podman specific flag, like `executor.podman.soci_artifact_store_target`. I don't see a lot of benefit of keeping this flag generic.

Reply:
Sure, sounds good